### DISCUSSION TITLE
Home Assistant OS 10.0

### DISCUSSION THREAD

**Discussion by agners** [home-assistant/operating-system#2473] [Announcements]
:information_source: Home Assistant OS 10.0 is currently on the beta channel only :information_source:

Home Assistant OS 10.0 is built from the latest Buildroot version 2023.02 and Linux kernel 6.1. This means that the major version of most packages got updated.

Most notable changes are:

- New board support: Hardkernel ODROID-M1
- zswap instead of swap in zram is used. This should allow to use Home Assistant OS on systems with lower amounts of RAM with the trade-off of slightly higher storage wear.
- Docker has been updated to the latest stable version 23.0.3.
- BlueZ version 5.66 with latest bug fixes
- NetworkManager 1.40.16 with bug fixes and improved IPv6 Neighbor Discovery Protocol support (required for Thread)
- OS-Agent 1.5.1 with new features for Supervisor (ready for improved data disk support/API to deploy SSH authentication keys)
- Firmware link support (Supervisor support will follow)
- Network file system mounting support (SMB/NFS, Supervisor support will follow)

## Changes

* Add label action for frontend issues (#2464) @agners
* Bump actions/stale from 7.0.0 to 8.0.0 (#2447) @dependabot
* updated generic_raw_uart and rpi-rf-mod to latest versions. (#2431) @jens-maus
* Add ODROID-M1 support (#2387) @agners
* Symlink firmware update directory to Supervisor writeable location (#2225) @agners
* Enable CONFIG_UHID for generic x86-64 based installs (#2369) @ANDuff
* updated generic-raw-uart and rpi-rf-mod package for kernel 6.1 compatibility (#2368) @jens-maus
* Bump ludeeus/action-shellcheck from 1.1.0 to 2.0.0 (#2331) @dependabot
* Bump actions/stale from 6.0.1 to 7.0.0 (#2289) @dependabot
* Update outdated ui references in issue template (#2175) @mdegat01
* Bump actions/stale from 6.0.0 to 6.0.1 (#2179) @dependabot
* Bump actions/stale from 5 to 6.0.0 (#2158) @dependabot
* Add comments, ignore vscode* (#2152) @ioctl2
* Remove UI reference from configuration USB flash drive docs (#2129) (#2131) @agners

## Home Assistant Operating System

* Add Broadcom NetXtreme/Tigon3 firmware (#2413) (#2466) @agners
* Add firmware for Realtek RTL8152/RTL8153 USB Ethernet Adapters (#2456) (#2465) @agners
* Bump buildroot to update Docker to 23.0.3 (#2452) @agners
* Use UTF-8 as default native language support (NLS) (#2450) @agners
* Add support for NFS and SMB/CIFS network file systems (#2446) @agners
* Enable Multi-Gen LRU (#2392) @agners
* Use zsmalloc for zswap (#2443) @agners
* Bump buildroot to update NetworkManager to 1.40.16 (#2442) @agners
* Bump buildroot to update Docker to 23.0.2 (#2439) @agners
* Set umask on swapfile creation (#2436) @agners
* Limit systemd-journald log size to 500MB (#2226) (#2435) @agners
* Improve IPv6 NDP for non-local Thread Border routers (#2434) @agners
* Fix swapfile creation for all memory sizes (#2427) @agners
* Don't kill ssh connection on OOM (#2424) @agners
* Use zswap instead of swap in zram (#2420) @agners
* Bump HAOS Agent to 1.5.1 (#2416) @agners
* Avoid moving data to same device (#2412) @agners
* Deactivate any external data disk device on first boot (#2390) (#2410) @agners
* Support systemd ConditionFirstBoot (#2409) @agners
* Choose Security Module using Kernel configuration (#2407) @agners
* Bump buildroot to fix sgdisk (#2404) @agners
* Add Multi-Gen LRU (disabled by default) (#2400) @agners
* Bump buildroot to 2023.02 (#2399) @agners
* Bump buildroot to 2023.02-rc3 (#2393) @agners
* Add libseccomp (#2389) @agners
* Use new containerd.sock location of Docker 23.0 (#2382) @agners
* Use Network device naming scheme v250 (#2380) @agners
* Enable NetworkManager command line utility nmcli explicitly (#2379) @agners
* Bump buildroot to update Docker and BlueZ (#2377) @agners
* Increase net.core.optmem_max for OTBR (#2375) @agners
* Bump buildroot to 2023.02-rc2 (#2374) @agners
* Revert NetworkManager bridge support (#2345) @agners
* Fix rtl8821cu build for Linux 5.15 (#2360) @agners
* NetworkManager: DHCPv4 handle lease NAK properly (#2341) (#2346) @agners
* Bump dbus-broker to 33 (#2340) @agners
* Add iproute2 ip command (#2335) @agners
* Bump buildroot to 2022.02.9 (#2313) @agners
* WiFi Access Point / HotSpot management in NetworkManager (#2304) @xonestonex
* Bump buildroot to add Bluetooth fixes (#2285) (#2286) @agners
* Bump buildroot to 2022.02.8 (#2278) @agners
* Use same /16 parent prefix for Docker as Supervisor (#2246) (#2259) @agners
* Bump buildroot to 2022.02.7 (#2253) @agners
* Enable experimental APIs for Bluetooth daemon (#2251) @agners
* Use iptables with nf_tables backend (#2250) @agners
* Decrease network size of Docker default bridge (#2135) (#2246) @agners
* Adding v4l2loopback device (#2243) @pergolafabio
* Add MLX5 support to OVA build (#2214) @rvdm82
* Disable Busybox' powertop (#2211) @agners
* Add fq_codel network packet scheduler algorithm by default (#2203) @agners
* Enable access to 'less' as a busybox applet (#2189) @ioctl2
* Bump buildroot to 2022.02.6 (#2198) @agners
* Add OS-Agent 1.4.1 support (#2157) @pvizeli
* Enable .config via /proc/config.gz + zcat via busybox (#2149) @ioctl2
* Connectivity check interval to 10 minutes (#2127) @mdegat01
* Fix Docker key.json corruption check (#2125) @agners
* Drop unused cgroup_enable=memory kernel parameter (#2124) @agners

## Build System

* Capitalize ODROID as used by Hardkernel (#2406) @agners
* Bump buildroot to 2023.02 (#2399) @agners
* Bump buildroot to 2023.02-rc3 (#2393) @agners
* Add label action for new features (#2388) @agners
* Avoid custom GPT location (#2386) @agners
* Support custom sized SPL/raw boot region (#2385) @agners
* Bump buildroot to 2023.02-rc2 (#2374) @agners
* Bump buildroot to fix build issue with sudo (#2347) (#2354) @agners
* Fix remote tracking branch for Buildroot submodule (#2287) (#2288) @agners
* Bump dessant/label-actions from 2 to 3 (#2263) @dependabot
* Enable release drafter on dev branch (#2252) @agners
* Replace set-output commands with GITHUB_OUTPUT environment vars (#2192) @agners

## Raspberry Pi

* Bump Raspberry Pi kernel and firmware to 1.20230405 (#2459) @agners
* Linux: Update kernel to 6.1.19 for Raspberry Pi systems (#2417) @agners
* Linux: Update kernel 6.1.13 for Raspberry Pi systems (#2373) @agners
* RaspberryPi: Update kernel 5.15.84 - 1.20230106 (#2312) @agners
* Add ASMedia based devices to UAS block list (#1743) (#2310) @agners
* Enable Realtek RTL8723AU/RTL8188[CR]U/RTL819[12][CE]U support (#2280) (#2281) @agners
* Bump U-Boot to v2023.01-rc1 (#2234) @agners
* Bump Raspberry Pi kernel and firmware to 1.20221104 (#2178) (#2236) @agners

## Home Assistant Yellow

* Fix build without device tree overlays (#2467) @agners
* Remove RPI-RF-MOD/HM-MOD-RPI-PCB hat overlays for Yellow (#2408) (#2461) @agners
* Linux: Update kernel to 6.1.19 for Raspberry Pi systems (#2417) @agners
* Linux: Update kernel 6.1.13 for Raspberry Pi systems (#2373) @agners
* Bump U-Boot to v2023.01 (#2316) @agners
* RaspberryPi: Update kernel 5.15.84 - 1.20230106 (#2312) @agners
* Enable Realtek RTL8723AU/RTL8188[CR]U/RTL819[12][CE]U support (#2280) (#2281) @agners
* Enable RTL8812AU WiFi support on additional boards (#2279) @agners
* Bump U-Boot to v2023.01-rc1 (#2234) @agners
* Bump Raspberry Pi kernel and firmware to 1.20221104 (#2178) (#2236) @agners
* Yellow: Support wipeing boot files on NVMe (#2173) @agners

## Open Virtual Appliance

* Linux: Update kernel 6.1.24 (#2469) @agners
* Partially revert "improve virtual hw support (#2138)" (#2425) @agners
* Bump QEMU guest agent to 7.2.0 (#2391) @agners
* Enable Microsoft HyperV integration services (#2381) @agners
* Linux: Update kernel 6.1.12 for non-Raspberry Pi systems (#2358) @agners
* Enable Realtek RTL8723AU/RTL8188[CR]U/RTL819[12][CE]U support (#2280) (#2281) @agners
* Enable RTL8812AU WiFi support on additional boards (#2279) @agners
* Add patch to avoid dummy waits on CPU idle for AMD systems (#2210) @agners
* Bump buildroot to fix GRUB2 build error (#2199) @agners
* Bump buildroot to fix CVE in open-vm-tools (#2195) @agners
* Add LXD VM Guest Agent Support (#2145) @bryanyork
* Allow multicast forwarding for IPv4 (#2160) @katter
* Fix proxmox mem hotplug, improve virtual hw support (#2138) @ioctl2
* Add Qualcomm QCA6174/QCA6174A Bluetooth/WiFi support (#2122) (#2123) @agners

## Generic x86-64

* Enable Broadcom Ethernet PHYs support (#2413) (#2470) @agners
* Linux: Update kernel 6.1.24 (#2469) @agners
* Add Broadcom NetXtreme/Tigon3 firmware (#2413) (#2466) @agners
* Linux: Update kernel 6.1.12 for non-Raspberry Pi systems (#2358) @agners
* Add RTL8723BE PCIe Wi-Fi chip support (#2329) (#2336) @agners
* Add amdgpu driver and firmware (#2320) @awilliam
* Add HP Smart Array driver (#2302) @d-two
* Enable Realtek RTL8723AU/RTL8188[CR]U/RTL819[12][CE]U support (#2280) (#2281) @agners
* Enable RTL8812AU WiFi support on additional boards (#2279) @agners
* Add patch to avoid dummy waits on CPU idle for AMD systems (#2210) @agners
* Enable Intel specific CPU idle driver (#2209) @agners
* Bump buildroot to fix GRUB2 build error (#2199) @agners
* Add Qualcomm QCA6174/QCA6174A Bluetooth/WiFi support (#2122) (#2123) @agners

## Hardkernel ODROID

* Linux: Update kernel 6.1.24 (#2469) @agners
* Document eMMC boot for ODROID-M1 (#2460) @agners
* Bump U-Boot for ODROID-M1 to 2023.04 (#2458) @agners
* Bump U-Boot for ODROID-M1 to 2023.04-rc4 (#2419) @agners
* Capitalize ODROID as used by Hardkernel (#2406) @agners
* Fix U-Boot environment size for ODROID-M1 (#2403) @agners
* Add ODROID-M1 to documentation (#2402) @agners
* Linux: Update kernel 6.1.12 for non-Raspberry Pi systems (#2358) @agners
* Bump U-Boot to v2023.01 for ODROID/Tinker/VIM3 (#2355) @agners
* Enable Realtek RTL8723AU/RTL8188[CR]U/RTL819[12][CE]U support (#2280) (#2281) @agners
* Bump U-Boot to v2023.01-rc1 (#2234) @agners
* Disable CONFIG_FW_LOADER_USER_HELPER(/_FALLBACK) (#2222) (#2229) @agners
* Fix S-ATA on ODROID-HC1 (#2153) (#2194) @agners
* Streamline CONFIG_RT_GROUP_SCHED (#2169) @pvizeli
* Remove HDMI fix (now part of Linux 5.15.68) (#2166) @agners
* Backport patch to fix ODROID-N2 HDMI (#2128) @agners

## ASUS Tinker

* Linux: Update kernel 6.1.24 (#2469) @agners
* Avoid custom GPT location (#2386) @agners
* Linux: Update kernel 6.1.12 for non-Raspberry Pi systems (#2358) @agners
* Bump U-Boot to v2023.01 for ODROID/Tinker/VIM3 (#2355) @agners
* Enable Realtek RTL8723AU/RTL8188[CR]U/RTL819[12][CE]U support (#2280) (#2281) @agners
* Bump U-Boot to v2023.01-rc1 (#2234) @agners

## Khadas VIM Series

* Linux: Update kernel 6.1.24 (#2469) @agners
* Linux: Update kernel 6.1.12 for non-Raspberry Pi systems (#2358) @agners
* Bump U-Boot to v2023.01 for ODROID/Tinker/VIM3 (#2355) @agners
* Enable Realtek RTL8723AU/RTL8188[CR]U/RTL819[12][CE]U support (#2280) (#2281) @agners
* Bump U-Boot to v2023.01-rc1 (#2234) @agners
* Disable CONFIG_FW_LOADER_USER_HELPER(/_FALLBACK) (#2222) (#2229) @agners
* Streamline CONFIG_RT_GROUP_SCHED (#2169) @pvizeli

## Generic aarch64

* Linux: Update kernel 6.1.24 (#2469) @agners
* Linux: Update kernel 6.1.12 for non-Raspberry Pi systems (#2358) @agners
* Enable Realtek RTL8723AU/RTL8188[CR]U/RTL819[12][CE]U support (#2280) (#2281) @agners
* Enable RTL8812AU WiFi support on additional boards (#2279) @agners
* Enable Hyper-V drivers for virtualization on ARM64 (#2256) (#2262) @agners
* Add QEMU Guest Agent to generic_aarch64 target (#2213) @lassebm
* Bump buildroot to fix GRUB2 build error (#2199) @agners
* Add Qualcomm QCA6174/QCA6174A Bluetooth/WiFi support (#2122) (#2123) @agners

<hr /><em>This discussion was created from the release <a href='https://github.com/home-assistant/operating-system/releases/tag/10.0'>Home Assistant OS 10.0</a>.</em>

**Comment by PenKlad** [comment]
Hi, Thanks for the great product and work everyone puts into it.
I just installed -Home Assistant OS 10.0- and my VLC media player stopped working.
What tests can I run to narrow down the cause?  (I rebooted HA) Thanks.
Note: Even the -Local Media- integration will not play locally stored .wav or.mp3 files.

Log:
Connection error: Failed to read: 0 bytes read on a total of undefined expected bytes. Partial bytes read: b''
11:21:08 AM – (ERROR) VLC media player via Telnet
System: 

    HA Blue, Odroid N2+ with NVMe SSD in USB enclosure running Home Assistant OS.
    USB speakers attached to HA Blue
    Radio Browser Integration
    Core-VLC Integration

**Reply by agners** [reply]
Is the log of the VLC add-on saying something specifically?

**Reply by PenKlad** [reply]
Not that I can tell.  Here is a pastie of the VLC log: http://pastie.org/p/5kFk8IiAmOIyV2qnun46nM
Note that even the -Local Media- integration does not output audio when trying to play locally stored *.mp3 or *.wav files.

**Reply by agners** [reply]
Which image/board are you using?

**Reply by PenKlad** [reply]
Thanks for staying with this.  I don't know what you mean. I just have VLC and Local Media integrations on HAOS running on an Odroid N2+ (HA Blue) and USB speakers plugged in to the HA Blue.

**Reply by PenKlad** [reply]
This latest update (Home Assistant 2023.4.6) fixed the USB speaker issue. Thanks

**Comment by cserem** [comment]
Odroid-M1 support finally! Thank you!

The documentation states that booting from nvme is not possible.
Is it on the roadmap, or it is too complicated to implement for the moment?

(The main selling point for me on the M1 was the m.2 slot, so I could finally replace the SD/EMMC with a -presumably - more reliable SSD storage.)

**Reply by agners** [reply]
You still can use it through the data disk feature. That moves pretty much everything which is commonly written to the NVMe. Combined with a eMMC you you'll have a very reliable platform.

The problem is the boot flow: We use our bootloader to provide a fallback OS installation. The ODROID-M1 provided boot loader (SPL which is part of petitboot) doesn't support booting from NVMe right now.

**Reply by apaneiro** [reply]
Odroid M1 would be a nice platform for HA + Frigate but, in this release, hassos doesn't seem to include the PCI drivers to run the M.2 coral device. I'm sure this will be fixed soon, the other thing missing is GPU acceleration, I haven't been able to install the Rockwell gpu drivers in debian.

**Comment by AarvdH** [comment]
Hi, Thank you for all your work on this. I have been waiting so long for the support on the OdriodM1. It wasn´t possible because it hat a RK3568B2 (Rockwell) chip on board. I just want to be sure, so now I can buy the OdroidM1 and I can get Homeassistant on it, So I mean the normal clean Homeassistant? No VM, no docker. (Sorry for my question, but I am not so good in these things. I have now the OdroidN2+ with eMMC and SSD vor Homeassistant. The same way I could now use the OdroidM1. Am I understanding this correctly? Because in the homeassistant.io site I don't see the image for it.  
Thank you very much and keep up the good work.

**Reply by agners** [reply]
Yes, you can use HAOS on ODROID-M1 just like on ODROID-N2+. There are some caveats currenlty, depending on what you'd like to boot from, see https://github.com/home-assistant/operating-system/blob/dev/Documentation/boards/hardkernel/odroid-m1.md.

> Because in the homeassistant.io site I don't see the image for it.

Documentation is still work in progress.

**Reply by AarvdH** [reply]
Thanks Odroid M1 is ordered.. Can't wait. One little other thing, I have the same problem as above with the VLC, But no USB speakers I have 3.5 mm jack and no sound anymore since 10.0. Log of VLC full of errors.

**Comment by simoncraddock** [comment]
Sadly this update didn't go well at all, I lost all my Zwave and Zigbee devices and couldn't get them back as both ZHA and Zigbee2MQTT services both failed to restart. I ended up going back to the previous version via nightly backup.

**Reply by sokai** [reply]
🤔 Which board/system? I made my upgrade on RPi4 and everything went very well …

**Reply by simoncraddock** [reply]
Generic x86-64.

Back on 9.5 currently

**Comment by ggian311** [comment]
First of all, thank you for all your hard work and for the great updates you give us every month.

Unfortunately, this update wasn't successful for me: it hanged a lot of times while doing the reboot. After twinkling with it a bit, I found out that the USB-connected devices (a Skyconnect and a Z-wave stick, were causing the boot process to get stuck in random places. 
After removing them, the boot process gets stuck always in the same spot, right after the sound card identification. 

Running on a laptop, intel CPU, 16gb Ram, sound card integrated on motherboard

**Reply by agners** [reply]
Thanks for you kind words.

Can you isolate which stick causes the problem? Or is it with both of them? I assume you are using the generic-x86-64 image? It keeps stuck even after ~5 minutes waiting? If this continues to be a problem you should file an issue in this repository.

If bootup isn't successful three times in a row the boot loader should revert back to the previous OS version.

**Reply by ggian311** [reply]
@agners  I am indeed using the generic-x86-64 image. With a couple more trials, I narrowed it down to the Skyconnect (not plugged in directly but with the appropriate cable to avoid interference; although I doubt this would hamper the boot process).
Situation:

- with Skyconnect --> random hangs at boot
- without Skyconnect --> boot stops right after soundcard 

The system gets stuck for more than 5 minutes and it correctly reverts to the previous OS version after the 3rd reboot (I'm not back on 9.5).

**Reply by agners** [reply]
> (I'm not back on 9.5).

"not" or did you meant to write "now"?

Now that you are back on HAOS 9.5, can you get the logs from that previous unsuccessful boot?

```
ha host logs -b -1 -n 10000
```

Best you create an issue with all information collected.

**Reply by ggian311** [reply]
Yes, I meant 'now'.
Will do, thanks!

**Reply by ggian311** [reply]
Thank you @agners. Issue opened [here](https://github.com/home-assistant/operating-system/issues/2502)

**Comment by megaz555** [comment]
This release is plagued by Bluetooth issues - see the issue report here:

https://github.com/home-assistant/operating-system/issues/2485
